### PR TITLE
Settings: detailed description bug exp1 WIP

### PR DESCRIPTION
**Pull request by cleidigh**
anchor problem when flipped above

When DD is above user moving selection would cause list to move up and down with different description sizes. (because variable element is on bottom between list and select anchor)
I don't think that's going to work well.  Description above will also look odd, but not have jumping.

I think we have to maintain below, and scroll the select if necessary
I am not aware of anything similar existing to compare this to.

your thoughts?  BTW the easiest way to see some of these things is to focus on the first select
and zoom then open the select when forced to flip above.  you can see some of the oddities.

I will continue to work on this tomorrow but I think I need your opinions

Update:  After more experimentation I propose the following:
- within selectbox , on open determine space , if insufficient below, fire move event
- within settings2 add fire move event handler, use tree.reveal(element, 0)
- do not open if cannot fit one entry

Basically we are working around limitations of contextV which does not to re-layout when above
and I have not found a clever way of forcing layout, perhaps there is a way.

### CODE REVIEW

**Comment by roblourens** [comment]
Hm, I guess I would avoid trying to move the settings editor around. You could have a dropdown at the very end of the list, and not be able to scroll it anywhere else. And we may always want to use the dropdown details somewhere else.

> When DD is above user moving selection would cause list to move up and down with different description sizes

Good point, I guess I hadn't thought about that case. Should we show the details on top then instead, in this scenario? It can have a variable height.

**Comment by cleidigh** [comment]
Addressed with new PR #58129
